avalon/tools/creator/app.py

```diff
@@ -20,6 +21,91 @@
 Separator = "---separator---"
 
 
+class SubsetNameValidator(QtGui.QRegExpValidator):
+
+    invalid = QtCore.Signal(set)
+    pattern = "^[a-zA-Z0-9_.]*$"
+
+    def __init__(self):
+        reg = QtCore.QRegExp(self.pattern)
+        super(SubsetNameValidator, self).__init__(reg)
+
+    def validate(self, input, pos):
+        results = super(SubsetNameValidator, self).validate(input, pos)
+        if results[0] == self.Invalid:
+            self.invalid.emit(self.invalid_chars(input))
+        return results
+
+    def invalid_chars(self, input):
+        invalid = set()
+        re_valid = re.compile(self.pattern)
+        for char in input:
+            if char == " ":
+                invalid.add("' '")
+                continue
+            if not re_valid.match(char):
+                invalid.add(char)
+        return invalid
+
+
+class SubsetNameLineEdit(QtWidgets.QLineEdit):
+
+    report = QtCore.Signal(str)
+    colors = {
+        "empty": (QtGui.QColor("#78879b"), ""),
+        "exists": (QtGui.QColor("#4E76BB"), "border-color: #4E76BB;"),
+        "new": (QtGui.QColor("#7AAB8F"), "border-color: #7AAB8F;"),
+    }
+
+    def __init__(self, *args, **kwargs):
+        super(SubsetNameLineEdit, self).__init__(*args, **kwargs)
+
+        validator = SubsetNameValidator()
+        self.setValidator(validator)
+        self.setToolTip("Only alphanumeric characters (A-Z a-z 0-9), "
+                        "'_' and '.' are allowed.")
+
+        self._cstatus = None
```

Comment:
Not sure if there's enough reason to abbrevaite `_color_status` to `_cstatus`. I don't mind it. Just wondering.

Reply:
Hmm, this should be improved for reading.